src/collector/mod.rs

```diff
@@ -60,6 +60,16 @@ impl From<CollectorOptions> for WebSocketListenerConfig {
 	}
 }
 
+/// Used to query storage for multiple hash types
+#[derive(Clone, Debug, Hash, Ord, PartialOrd, Eq, PartialEq)]
+pub(crate) struct CollectorKey {
+	pub prefix: String,
+	pub hash: Option<H256>,
```

Comment:
why is this optional ?

Reply:
For prefixed searches, as `None < Some(H256)`.